doc/src/modules/polys/domainmatrix.rst

```diff
@@ -0,0 +1,48 @@
+.. _polys-domainmatrix:
+
+===============================================
+Introducing the domainmatrix of the poly module
+===============================================
+
+This page introduces the idea behind domainmatrix which is used in SymPy's
+:mod:`sympy.polys` module. This is a relatively advanced topic so for a better understanding
+it is recommended to read about :py:class:`~.Domain` and :py:class:`~.DDM` along with
+:mod:`sympy.matrices` module.
+
+What is domainmatrix?
+=====================
+
+It is way of associating Matrix with :py:class:`~.Domain`.
+
+A domainmatrix represents a matrix with elements that are in a particular
+Domain. Each domainmatrix internally wraps a DDM which is used for the lower-level operations.
```

Comment:
This is not actually true:
```python
In [10]: dM
Out[10]: DomainMatrix([[1, 2, 3], [4, 5, 6], [7, 8, 9]], (3, 3), ZZ)

In [11]: dM.rep
Out[11]: 
{0: {0: mpz(1), 1: mpz(2), 2: mpz(3)},
 1: {0: mpz(4), 1: mpz(5), 2: mpz(6)},
 2: {0: mpz(7), 1: mpz(8), 2: mpz(9)}}

In [12]: type(dM.rep)
Out[12]: sympy.polys.matrices.sdm.SDM
```
The internal representation can be `DDM` or `SDM`.

(Right now it defaults to `SDM` but that could change so I wouldn't mention that here.)

Reply:
The SDM class should be added to the docs as well.

Reply:
Ok I'll fix that and just leave it at `A domainmatrix represents a matrix with elements that are in a particular
Domain.`

Reply:
I think it's good to mention `DDM` and `SDM`. We just need to make sure that the description is accurate.